misc/python/materialize/output_consistency/input_data/operations/number_operations_provider.py

```diff
@@ -0,0 +1,461 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from typing import List
+
+from materialize.output_consistency.data_type.data_type_category import DataTypeCategory
+from materialize.output_consistency.expression.expression_characteristics import (
+    ExpressionCharacteristics,
+)
+from materialize.output_consistency.input_data.params.number_operation_param import (
+    NumericOperationParam,
+)
+from materialize.output_consistency.input_data.validators.number_args_validator import (
+    MaxMinusNegMaxArgsValidator,
+    MultiParamValueGrowsArgsValidator,
+    SingleParamValueGrowsArgsValidator,
+    Uint8MixedWithTypedArgsValidator,
+)
+from materialize.output_consistency.operation.operation import (
+    DbFunction,
+    DbOperation,
+    DbOperationOrFunction,
+)
+
+NUMERIC_OPERATION_TYPES: List[DbOperationOrFunction] = []
+
+# ===== BEGIN NUMBER OPERATORS =====
+
+NUMERIC_OPERATION_TYPES.append(
+    DbOperation(
+        "$ + $",
+        [NumericOperationParam(), NumericOperationParam()],
+        DataTypeCategory.NUMERIC,
+        {MultiParamValueGrowsArgsValidator()},
+    )
+)
+NUMERIC_OPERATION_TYPES.append(
+    DbOperation(
+        "$ - $",
+        [NumericOperationParam(), NumericOperationParam()],
+        DataTypeCategory.NUMERIC,
+        {MaxMinusNegMaxArgsValidator()},
+    )
+)
+NUMERIC_OPERATION_TYPES.append(
+    DbOperation(
+        "$ * $",
+        [NumericOperationParam(), NumericOperationParam()],
+        DataTypeCategory.NUMERIC,
+        {MultiParamValueGrowsArgsValidator()},
+    )
+)
+NUMERIC_OPERATION_TYPES.append(
+    DbOperation(
+        "$ / $",
+        [
+            NumericOperationParam(),
+            NumericOperationParam(incompatibilities={ExpressionCharacteristics.ZERO}),
+        ],
+        DataTypeCategory.NUMERIC,
+    )
+)
+NUMERIC_OPERATION_TYPES.append(
+    DbOperation(
+        "$ % $",
+        [
+            NumericOperationParam(),
+            NumericOperationParam(incompatibilities={ExpressionCharacteristics.ZERO}),
+        ],
+        DataTypeCategory.NUMERIC,
+    )
+)
+# Bitwise AND
```

Comment:
All the bitwise operators are low relevance I think.

Reply:
✅